File: test-utility/src/main/java/com/mobilesolutionworks/android/controller/test/util/WaitForIdle.java
package com.mobilesolutionworks.android.controller.test.util;

import android.support.test.espresso.UiController;
import android.view.View;

/**
 * Created by yunarta on 12/3/17.
 */

public class WaitForIdle extends PerformRootAction {

    private int mMillisDelay;

    public WaitForIdle() {
        this(100);
    }

    public WaitForIdle(int millisDelay) {
        mMillisDelay = millisDelay;
    }

    @Override
    public void perform(UiController uiController, View view) {
        uiController.loopMainThreadForAtLeast(mMillisDelay);
        uiController.loopMainThreadUntilIdle();
    }
}

File: test-app/src/androidTest/kotlin/com/mobilesolutionworks/android/controller/test/util/WaitForIdle.kt
package com.mobilesolutionworks.android.controller.test.util

import android.support.test.espresso.UiController
import android.view.View

/**
 * Created by yunarta on 12/3/17.
 */

class WaitForIdle @JvmOverloads constructor(private val mMillisDelay: Int = 100) : PerformRootAction() {

    override fun perform(uiController: UiController, view: View) {
        uiController.loopMainThreadForAtLeast(mMillisDelay.toLong())
        uiController.loopMainThreadUntilIdle()
    }
}
